### PR TITLE
Docker swarm features

### PR DESCRIPTION
**Pull request by magrossi**
Docker swarm low level features section of the report.

### CODE REVIEW

File: report/is2006_latex_template.tex [hunk 1]
```diff
@@ -115,14 +114,34 @@ \section{Introduction}
 
 \section{Study Design}
 \lipsum[1]
+
+\subsection{Docker Swarm}
+A part of the docker's native ecosystem, swarm is docker's production-ready cluster management and orchestration tool. It allows for transparent scaling of docker containers across multiple hosts and because it serves the same docker daemon API it can be used seamlessly with any docker compatible tools (like Docker Compose, Dokky, Jenkins and others).
```
Comment (full-of-foo) [inline report/is2006_latex_template.tex:119]: Capital 'S' and 'D' needed here

File: report/is2006_latex_template.tex [hunk 2]
```diff
@@ -115,14 +114,34 @@ \section{Introduction}
...
+Swarm will run a swarm agent daemon in each node of the cluster and coordinate them using a pluggable discovery service back-end (which maintains a list of nodes in your cluster) and a swarm manager (that will actually control the whole cluster). The connection between the manager and agent nodes is made through a secure tcp connection (by means of TSL certificates). In previous versions of swarm the manager itself was the single point of failure of the cluster as there was no fail-over available. In the latest release however, version 1.0, a replication service for the manager has been developed whereby several manager instances will compete for leadership, making swarm a highly available solution.
```
Comment (full-of-foo) [inline report/is2006_latex_template.tex:120]: Perhaps a citation is needed here

File: report/is2006_latex_template.tex [hunk 3]
```diff
@@ -115,14 +114,34 @@ \section{Introduction}
...
+\item \textbf{Port} considers exposed ports on the node as unique resources, and will try to schedule containers on nodes that have a particular port available.
+\item \textbf{Dependency} will try to honour volume, network stack or link dependencies and schedule containers only on nodes that match.
+\item \textbf{Health} will prevent scheduling of new containers on unhealthy nodes.
 \end{itemize}
+\textbf{*} Both constraint and affinity filters are hard enforced, which means that if the affinity or constraint is not met, the container will not be started. The user has an option to create the affinity or constraint in soft mode, which will make the scheduler disregard the filter and use the configured strategy in case the affinity or constraint is not met.
+\subsubsection{Not yet implemented}
+Swarm does not yet support (as of version 1.0) host rebalancing - where a host fails and needs to be rescheduled on another node. This is in the roadmap and should be released alongside Docker version 1.10 release [https://github.com/docker/swarm/wiki/0.6.0-Milestone-Project-Page].
```
Comment (full-of-foo) [inline report/is2006_latex_template.tex:141]: The link reference here needs to be Harvard style

**Comment by full-of-foo** [comment]
I will have a closer look tomorrow and merge

**Comment by magrossi** [comment]
Thanks Tony!

Sorry, I'm not very experienced with LateX I'm afraid :(

**Pull request merged**